cmd/object-scaffold/main.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/controller-tools/pkg/object"
+)
+
+func main() {
+	opts := &object.ScaffoldOptions{
+		Resource: object.Resource{
+			Namespaced: true,
+		},
+	}
+	scaffoldCmd := &cobra.Command{
+		Use:   "object-scaffold",
+		Short: "Quickly scaffold out basic bits of a Kubernetes object.",
+		Long: `Quickly scaffold out the structure of a Kubernetes root type and associated types.
```

Comment:
May be take out `root` out ? When I read for the first time, I didn't get it. Then I realized, we are referring to the object type. 

Thinking more... an alternative good name for the command could be `type-scaffold` ?